helix-term/src/ui/editor.rs

```diff
@@ -446,12 +446,16 @@ impl EditorView {
             .map(|range| range.cursor_line(text))
             .collect();
 
-        let mut breakpoints: Option<Vec<SourceBreakpoint>> = None;
-        let mut stack_pointer: Option<StackFrame> = None;
+        let mut breakpoints: Option<&Vec<SourceBreakpoint>> = None;
+        let mut stack_frame: Option<&StackFrame> = None;
         if let Some(debugger) = debugger {
             if let Some(path) = doc.path() {
-                breakpoints = debugger.breakpoints.get(path).cloned();
-                stack_pointer = debugger.stack_pointer.clone()
+                breakpoints = debugger.breakpoints.get(path);
+
+                if let (Some(frame), Some(thread_id)) = (debugger.active_frame, debugger.thread_id)
+                {
+                    stack_frame = debugger.stack_frames[&thread_id].get(frame); // TODO: drop the clone..
```

Comment:
This introduced a crash I'm currently fixing.
- Launch
- Set 2 breakpoints
- Run
- Continue
- Editor crashed